pkg/services/policer/check.go

```diff
@@ -106,31 +118,65 @@ func (p *Policer) processObject(ctx context.Context, addrWithType objectcore.Add
 	}
 
 	c := &processPlacementContext{
-		Context: ctx,
+		Context:      ctx,
+		object:       addrWithType,
+		checkedNodes: newNodeCache(),
 	}
 
-	var numOfContainerNodes int
-	for i := range nn {
-		numOfContainerNodes += len(nn[i])
-	}
-
-	// cached info about already checked nodes
-	checkedNodes := newNodeCache()
-
 	for i := range nn {
 		select {
 		case <-ctx.Done():
 			return
 		default:
 		}
 
-		p.processNodes(c, addrWithType, nn[i], policy.ReplicaNumberByIndex(i), checkedNodes)
+		p.processNodes(c, nn[i], policy.ReplicaNumberByIndex(i))
+	}
+
+	// if context is done, needLocalCopy might not be able to calculate
+	select {
+	case <-ctx.Done():
+		return
+	default:
 	}
 
 	if !c.needLocalCopy {
-		p.log.Info("redundant local object copy detected",
-			zap.Stringer("object", addr),
-		)
+		if !c.localNodeInContainer {
+			// Here we may encounter a special case where the node is not in the network
+			// map. In this scenario, it is impossible to determine whether the local node
+			// will enter the container in the future or not. At the same time, the rest of
+			// the network will perceive local peer as a 3rd party which will cause possible
+			// replication problems. Iin order to avoid the potential loss of a single
+			// replica, it is held.
+			if !p.network.IsLocalNodeInNetmap() {
```

Comment:
BTW, maybe this logic could reside somewhere in the upper layer (that calls `processObject()` at least), because in this case checking individual objects seems to be useless, we won't touch them anyway.

Reply:
Maybe, but then we won't try to replicate them into the actual container.

Reply:
But we can't replicate anyway unless we're a part of the current netmap?